File: fix_booking_statuses.js
// Script to fix booking statuses - mark future bookings as confirmed instead of completed
// Run this with: node fix_booking_statuses.js

const admin = require('firebase-admin');

// Initialize Firebase Admin SDK
const serviceAccount = require('./path/to/serviceAccountKey.json');  // Update this path

admin.initializeApp({
  credential: admin.credential.cert(serviceAccount)
});

const db = admin.firestore();

async function fixBookingStatuses() {
  try {
    console.log('🔍 Fetching all bookings with completed status...');
    
    const bookingsSnapshot = await db.collection('bookings')
      .where('status', '==', 'completed')
      .get();
    
    console.log(`Found ${bookingsSnapshot.size} completed bookings`);
    
    const now = new Date();
    let updatedCount = 0;
    
    for (const doc of bookingsSnapshot.docs) {
      const data = doc.data();
      const bookingId = doc.id;
      
      let bookingDateTime = null;
      
      // Parse booking date
      if (data.bookingDate && data.bookingDate._seconds) {
        // Timestamp format
        bookingDateTime = new Date(data.bookingDate._seconds * 1000);
      } else if (typeof data.bookingDate === 'string') {
        // String format (M/d/yyyy)
        const parts = data.bookingDate.split('/');
        if (parts.length === 3) {
          bookingDateTime = new Date(parts[2], parts[0] - 1, parts[1]);
          
          // Add time if available
          if (data.timeSlot) {
            const timeParts = data.timeSlot.split(':');
            if (timeParts.length >= 2) {
              bookingDateTime.setHours(parseInt(timeParts[0]), parseInt(timeParts[1]));
            }
          }
        }
      }
      
      if (bookingDateTime && bookingDateTime > now) {
        console.log(`📅 Fixing future booking ${bookingId}: ${bookingDateTime.toISOString()}`);
        await doc.ref.update({ status: 'confirmed' });
        updatedCount++;
      }
    }
    
    console.log(`✅ Updated ${updatedCount} future bookings to confirmed status`);
    process.exit(0);
  } catch (error) {
    console.error('❌ Error:', error);
    process.exit(1);
  }
}

fixBookingStatuses();

File: lib/utils/fix_booking_statuses.dart
import 'package:cloud_firestore/cloud_firestore.dart';
import 'package:flutter/material.dart';

/// Utility to fix booking statuses - marks future bookings as confirmed instead of completed
class BookingStatusFixer {
  final FirebaseFirestore _firestore;

  BookingStatusFixer(this._firestore);

  /// Fix all future bookings that were incorrectly marked as completed
  Future<int> fixFutureBookings() async {
    try {
      debugPrint('🔍 Fetching all bookings with completed status...');

      final snapshot = await _firestore
          .collection('bookings')
          .where('status', isEqualTo: 'completed')
          .get();

      debugPrint('Found ${snapshot.docs.length} completed bookings');

      final now = DateTime.now();
      int updatedCount = 0;

      for (var doc in snapshot.docs) {
        final data = doc.data();
        final bookingId = doc.id;

        DateTime? bookingDateTime;

        // Parse booking date
        final bookingDateField = data['bookingDate'];

        if (bookingDateField is Timestamp) {
          bookingDateTime = bookingDateField.toDate();
        } else if (bookingDateField is String) {
          try {
            final parts = bookingDateField.split('/');
            if (parts.length == 3) {
              final month = int.parse(parts[0]);
              final day = int.parse(parts[1]);
              final year = int.parse(parts[2]);

              bookingDateTime = DateTime(year, month, day);

              // Add time if available
              final timeSlot = data['timeSlot'] as String?;
              if (timeSlot != null) {
                final timeParts = timeSlot.split(':');
                if (timeParts.length >= 2) {
                  final hour = int.tryParse(timeParts[0]) ?? 0;
                  final minute = int.tryParse(timeParts[1]) ?? 0;
                  bookingDateTime = DateTime(year, month, day, hour, minute);
                }
              }
            }
          } catch (e) {
            debugPrint('Error parsing booking date: $bookingDateField - $e');
          }
        }

        // If booking is in the future, change status to confirmed
        if (bookingDateTime != null && bookingDateTime.isAfter(now)) {
          debugPrint('📅 Fixing future booking $bookingId: $bookingDateTime');
          await _firestore.collection('bookings').doc(bookingId).update({
            'status': 'confirmed',
          });
          updatedCount++;
        }
      }

      debugPrint('✅ Updated $updatedCount future bookings to confirmed status');
      return updatedCount;
    } catch (e) {
      debugPrint('❌ Error fixing booking statuses: $e');
      rethrow;
    }
  }
}
